packages/playwright-core/src/server/browserContext.ts

```diff
@@ -688,6 +702,7 @@ export function validateBrowserContextOptions(options: channels.BrowserNewContex
     options.proxy = normalizeProxySettings(options.proxy);
   }
   verifyGeolocation(options.geolocation);
+  verifyClientCertificates(options.clientCertificates);
```

Comment:
I think it makes sense to move this line into `createClientCertificatesProxyIfNeeded` - otherwise we'll start the proxy with malformed options.